app/presenters/queries/content_item_presenter.rb

```diff
@@ -24,20 +26,43 @@ def initialize(content_item_scope, fields = nil)
       end
 
       def present
+        group_items(groups).compact
+      end
+
+    private
+
+      attr_accessor :content_item_scope, :fields
+
+      def groups
         scope = join_supporting_objects(content_item_scope)
         scope = select_fields(scope).order(order)
 
-        items = ActiveRecord::Base.connection.execute(scope.to_sql)
-        groups = items.group_by { |i| [i.fetch("content_id"), i.fetch("locale")] }
+        ActiveRecord::Base.connection.execute(aggregated_sql(scope.to_sql))
+      end
+
+      def aggregated_sql(sql)
+        <<-END.strip_heredoc
+          SELECT json_agg(json_rows) FROM (
+            SELECT row_to_json(item) json_item FROM (#{sql}) item
+          ) json_rows
+          GROUP BY json_item->>'content_id', json_item->>'locale'
+          ORDER BY max(json_item->>'public_updated_at') DESC
```

Comment:
Since `order` is a parameter we should interpolate it here rather than hard-coding.